### DISCUSSION TITLE
Impossible to fulfill assert in Notice 1, in OPT-201-Organization-TouchPoint

### DISCUSSION THREAD

**Discussion by PiotrRymarz** [OP-TED/eForms-SDK#884] [Q&A] [answered]
Hi,
as far as I understand it is impossible to correctly fill notice 1.

We have field OPT-201-Organization-TouchPoint which has hidden ID (TPO-XXXX). This means that we have to fill value in, because end user does not have such possibility. So we automatically have TPO-0001 in our notice. It is also possible for end user to add additional touch points, so we have additional TPO's (TPO-0002, TPO-0003 and so forth)

With this field there is an assertions
```
{
        "condition" : "{ND-Organization} ${OPT-201-Organization-TouchPoint is present}",
        "value" : "{ND-Organization} ${((OPT-201-Organization-TouchPoint in OPT-301-Lot-AddInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-DocProvider) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-EmployLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-EnvironLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-FiscalLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-Mediator) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-ReviewInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-ReviewOrg) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-TenderEval) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-TenderReceipt) or (OPT-201-Organization-TouchPoint in OPT-301-Part-AddInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Part-DocProvider) or (OPT-201-Organization-TouchPoint in OPT-301-Part-EmployLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Part-EnvironLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Part-FiscalLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Part-Mediator) or (OPT-201-Organization-TouchPoint in OPT-301-Part-ReviewInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Part-ReviewOrg) or (OPT-201-Organization-TouchPoint in OPT-301-Part-TenderEval) or (OPT-201-Organization-TouchPoint in OPT-301-Part-TenderReceipt) or (OPT-201-Organization-TouchPoint in OPT-301-ReviewBody) or (OPT-201-Organization-TouchPoint in OPT-301-ReviewReq))}",
        "severity" : "ERROR",
        "message" : "rule|text|BR-OPT-00201-0101"
      }
```
      
      which basically says, that if we have any TPO value (which we are forced to have) it must be used in one of many listed fields.
      
      And here comes the issue.
     Non of listed fields is present in notice 1. So from one side we are forced to add TPO-XXXX ids, and from other we are forced to fill fields which are not present.
     
     Am I maybe missing something?

**Comment by YvesJo** [comment, marked as answer]
Hi,
All the roles that may be associated to Touchpoints are at the Lot level. PIN profile notices do not have lots and therefore touchpoints are useless for these.
Also, for other notices, the use of touchpoints is not always needed.
The rule checks that any touchpoint is associated to at least a role it may be associated to.
When the user does not need touchpoints, he will not fill the corresponding fields and the associated structure should not be present in the XML. 
At XML generation, given the adopted approach, it is not sufficient to check for the existence of a value for OPT-201-Organization-TouchPoint (as it is set by the application) and it should be checked that further touchpoint information exists before creating the corresponding XML fragment.

**Reply by kpankowski** [reply]
@YvesJo Shouldn't all fields under the "GR-Touch-Point" group then be marked as forbidden for PIN notices that do not have LOTs or PARs?

Do you know if it's safe to assume that we can skip creating an XML node (and its parent) for a hidden field that also has the "_idScheme" attribute when all other fields in the same group are not filled by the user? Or perhaps only the TPO should be treated this way?

**Reply by YvesJo** [reply]
Hi,
idScheme may exist for any XML element of (XML) type identifier and not all fields associated to such elements are technical identifiers for an object, they could be references instead. The ones automatically managed by the application could be good candidates though.
